File: solutions/cpp/pangram/1/pangram.cpp
#include "pangram.h"

namespace pangram {

    bool is_pangram(std::string phrase) {
        if ( phrase == "" ) { return false; }

        for ( std::string::size_type i = 0; i < phrase.size(); i++ ) {
            phrase[i] = tolower(phrase[i]);
        }

        std::string alpha = "abcdefghijklmnopqrstuvwxyz";
        std::string::size_type count = 0;
        for ( std::string::size_type i = 0; i < alpha.size(); i++ ) {
            if ( phrase.find(alpha[i]) != std::string::npos ) { count++; }
        }
        bool answer = false;
        if ( count == alpha.size() ) { answer = true; }

        return answer;
    }

}  // namespace pangram

File: solutions/fortran/pangram/1/pangram.f90
module pangram
  implicit none
contains

  logical function is_pangram(sentence)
    character(*)      :: sentence
    character(len=len(sentence)) :: string
    character(len=26) :: alpha
    integer i, c, count

    print *, 'SENTENCE: ' // sentence

    do i = 1, len(sentence)
       c = iachar(sentence(i:i))
       if ( c >= 65 .and. c <= 90 ) then
          string(i:i) = achar(c + iachar(' '))
       else
          string(i:i) = sentence(i:i)
       end if
    end do

    alpha = 'abcdefghijklmnopqrstuvwxyz'
    count = 0
    do i = 1, len(alpha)
       if ( index(string, alpha(i:i)) > 0 ) count = count + 1
    end do

    is_pangram = count == len(alpha)

   end function is_pangram

end module pangram
